service/orchestrator/metrics_test.go

```diff
@@ -224,9 +224,9 @@ compliant {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			gotImpl, err := loadMetricImplementation(tt.args.metricID, tt.args.file)
-
-			assert.NoError(t, gotImpl.Validate())
-			tt.wantErr(t, err)
+			if tt.wantErr(t, err) && err == nil {
```

Comment:
You can also remove the "err== nil" in the following checks.

Reply:
If is necessary because even if no error is present, tt.wantErr() will return true (because assert.NoError will return true). But we only want to check the validation if no error was present.

Reply:
After discussion, we will leave it as is.